### PR TITLE
Items disappearing from View Sent Form

### LINKED ISSUE #1110: Items disappearing from View Sent Form
#### Software and hardware versions 
Collect v1.7.0, Android v7.1.2, Nexus 6P

#### Problem description
Items disappearing from View Sent Form screen upon filling out a new instance.

#### Steps to reproduce the problem
1. Fill out instance.
1. Send to server.
1. Delete all saved forms via Delete Saved Form.
1. Confirm shows as deleted in View Sent Form.
1. Fill out another instance.
1. Go back to View Sent Form, first instance has disappeared.

#### Expected behavior
Deleted instances should remain viewable under View Sent Form (with crossed eye icon) forever (?) I am not sure of the exact expected behavior as this feature is fairly new to me but @lognaturel said this isn't it :)

### PR DESCRIPTION
**Pull request by grzesiek2010**
This PR is in reference to resolve #1110

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java [hunk 1]
```diff
@@ -107,7 +107,9 @@ protected String doInBackground(Void... params) {
                     while (instanceCursor.moveToNext()) {
                         String instanceFilename = instanceCursor.getString(
                                 instanceCursor.getColumnIndex(InstanceColumns.INSTANCE_FILE_PATH));
-                        if (candidateInstances.contains(instanceFilename)) {
+                        String instanceStatus = instanceCursor.getString(
```
Comment (nribeka) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java:110]: @grzesiek2010 how about we add the status restriction in the query? That way we will get less records from the db and we don't need to do iteration here.

**Comment by grzesiek2010** [comment]
@nribeka we can't do that please notice that [candidateInstance](https://github.com/opendatakit/collect/pull/1126/files#diff-c6b798e69cb47b6c50f0e791fb67ab1dL85) contains all instances (those submitted as well). So we need to have a cursor that contains all to remove submitted instances https://github.com/opendatakit/collect/pull/1126/files#diff-c6b798e69cb47b6c50f0e791fb67ab1dR113

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java:110]: @nribeka we can't do that please notice that candidateInstance contains all instances (those submitted as well) https://github.com/opendatakit/collect/pull/1126/files#diff-c6b798e69cb47b6c50f0e791fb67ab1dL85. So we need to have a cursor that contains all to remove submitted instances https://github.com/opendatakit/collect/pull/1126/files#diff-c6b798e69cb47b6c50f0e791fb67ab1dR113 from the collection

Comment (nribeka) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java:110]: @grzesiek2010 you're right. The sync used everything in the folder and we need to go through all records in the db to make sure we don't add duplicates and removed dangling records.

**Pull request merged**